test/distributed/_tensor/test_utils.py

```diff
@@ -119,5 +123,82 @@ def test_compute_local_shape_and_global_offset_2D(self):
                 )
 
 
+class Test2DStridedLocalShard(DTensorTestBase):
+    @property
+    def world_size(self):
+        return 4
+
+    @with_comms
+    def test_fsdp1_tp_2d_dtensor_local_shards_and_offsets(self):
+        # We are mimicking the behavior of FSDP1 + TP.
+        # Currently, the 2D DTensor's local shard is correct, since from_local + redistribute incurs a all_gather behind the scene.
+        # When we have a global_tensor of [0, 1, 2, 3, 4, 5, 6, 7], the local shard of 2D DTensor would be:
+        # rank0: [0, 1], rank1: [2, 3], rank2: [4, 5], rank3: [6, 7]
+        comm_mode = CommDebugMode()
+        with comm_mode:
```

Comment:
nit: Can this be `with CommDebugMode() as comm_mode:`?

Reply:
Good catch. Definitely was not thinking lol.

Reply:
Oops realized I do actually need `comm_mode` for checking the number of collective call later on. Changing it back.

Reply:
If we do `with CommDebugMode() as comm_mode`, I thought we can still access `comm_mode`.